src/liballoc/collections/btree/set.rs

```diff
@@ -109,6 +109,80 @@ pub struct Range<'a, T: 'a> {
     iter: btree_map::Range<'a, T, ()>,
 }
 
+/// Core of an aLternative to Peekable, that is more efficient
+/// if copying items is cheap and if peeking early does no harm.
+/// It always has one item read ahead.
+#[derive(Clone)]
+struct Peeking<I>
+where
+    I: Iterator,
+    I::Item: Copy,
+{
+    head: Option<I::Item>,
+    tail: I,
+}
+impl<I> Peeking<I>
+where
+    I: ExactSizeIterator,
+    I::Item: Copy,
+{
+    fn new(mut iter: I) -> Self {
+        let head = iter.next();
+        Peeking { head, tail: iter }
+    }
```

Comment:
nitpicks, but empty lines between methods would be good style as well (See the rest of the file)

Reply:
Well, I mimicked the whitespace of Iter and IntoIter... but it seems they were written during a whitespace crisis.